src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,192 @@
+              </UnorderedList>
+            );
+          } else if (courseDetails) {
+            // Extract course details and format credits
+            const pid = courseDetails.pid;
+            const credits = courseDetails.credits.credits;
+            const creditsVisual =
+              credits.min === credits.max ? `(${credits.max})` : `(${credits.min} - ${credits.max})`;
+
+            // Render course details with a hyperlink to requisite course page on courseup
+            return (
+              <UnorderedList>
+                <ListItem title={`${subject} ${code}`} style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                  <Text _hover={{ color: 'blue.600' }} color="blue.400" as="span">
+                    <Link to={`/calendar/${currTerm}/${subject}?pid=${pid}`}>{`${subject} ${code}`}</Link>
+                  </Text>
+                  {` - ${courseDetails.title} ${creditsVisual}`}
+                </ListItem>
+              </UnorderedList>
+            );
+          } else {
+            // Render default state
+            return (
+              <UnorderedList>
+                <ListItem
+                  title={`${subject} ${code}`}
+                  style={{ marginLeft: `${indentationLevel * 40}px` }}
+                >{`${subject} ${code}`}</ListItem>
+              </UnorderedList>
+            );
+          }
+        }}
+      </GetCourseDetails>
+    );
+  }
+  return <></>;
+}
+
+// Get nested reqs from NestedPreCoRequisites object
+export function nestedReq(req: NestedPreCoRequisites): Array<string | NestedPreCoRequisites | KualiCourse> {
+  if (req.reqList !== undefined) {
+    return req.reqList;
+  }
+  return [];
+}
+
+// Defines the properties of the Requisites component;
+// an array of strings, NestedPreCoRequisites and KualiCourse
+export type RequisiteProp = {
+  preAndCorequisites?: (string | NestedPreCoRequisites | KualiCourse)[];
+  preOrCorequisites?: (string | NestedPreCoRequisites | KualiCourse)[];
+};
+
+// Renders the prerequisites and corequisites as a JSX element
+export function Requisites({ preAndCorequisites, preOrCorequisites }: RequisiteProp) {
+  return (
+    <Box>
+      {/* if preAndCorequisites are provided, render them with a heading */}
+      {preAndCorequisites && (
+        <Box>
+          Prerequisites
+          <Divider />
+          {/* call the DisplayRequirement for each prerequisite */}
+          {preAndCorequisites.map((req) => (
+            <Box>
+              <UnorderedList>{DisplayRequirement(req)}</UnorderedList>
+            </Box>
+          ))}
+        </Box>
+      )}
+      {/* if preOrCorequisites are provided, render them with a heading */}
+      {preOrCorequisites && (
+        <Box>
+          Pre Or Corequisites
+          <Divider />
+          {/* call the DisplayRequirement for each prerequisite */}
+          {preOrCorequisites.map((req) => (
+            <Box>
+              <UnorderedList>{DisplayRequirement(req)}</UnorderedList>
```

Comment:
Never seen recursive react components, nicely done 😎

Reply:
Thank you 😄 
This was definitely the thing holding me back initially